File: exercicios/matriz-transposta/matriz-transposta.js
'use strict';

module.exports = {
  matriz_transposta: function (l, c, matriz) {
    var i, j, aux;

    for (i = 0; i < l; i++) {
      for (j = i + 1; j < c; j++) {
        if (j !== i) {
          aux = matriz[i][j];
          matriz[i][j] = matriz[j][i];
          matriz[j][i] = aux;
        }
      }
    }
    return matriz;
  },
  generateRandom: function (min, max) {
    return Math.floor(Math.random() * (max - min + 1)) + min;
  },
  countElements: function (linhas, colunas, matriz) {
    var linhaCount, colunaCount, elementCount = 0;
    for (linhaCount = 0; linhaCount < linhas; linhaCount++) {
      for (colunaCount = 0; colunaCount < colunas; colunaCount++) {
        elementCount++;
      }
    }
    return elementCount;
  },
  countLines: function (matriz) {
    var linhaCount, colunaCount;
    for(linhaCount = 0; linhaCount < matriz.length; linhaCount++){
       for(colunaCount = 0; colunaCount < matriz[linhaCount].length; colunaCount++){
        // colunaCount++;
        // console.log('linha %d coluna %d', linhaCount, colunaCount);
        // console.log('colunaCount', colunaCount);
       }
    }

    return linhaCount;
  },
  countColumns: function (matriz) {
    return matriz.length;
  }
};
// module.exports = module;

File: exercicios/matriz-transposta/matriz-transposta.c
#include <stdio.h>
 
void matriz_transposta (int l, int c, int matriz[l][c]) {
  int i, j, aux;
   
  for (i = 0; i < l; i++) {
    for (j = i+1; j < c; j++) {
      if (j != i) {
   aux = matriz[i][j];
   matriz[i][j] = matriz[j][i];
   matriz[j][i] = aux;
      }
    }
  }
   
  for (i = 0; i < l; i++) {
    for (j = 0; j < c; j++) {
      printf ("%d ",matriz[i][j]);
    }
    printf ("\n");
  }
   
}
 
main () {
  int l, c, i, j;
   
  scanf ("%d%d",&l, &c);
   
  int matriz[l][c];
   
  for (i = 0; i < l; i++) {
    for (j= 0; j < c; j++) {
      scanf ("%d",&matriz[i][j]);
    }
  }
  printf ("\n");
   
  matriz_transposta (l, c, matriz);
   
}